### ISSUE TITLE
[Bug]: Cannot set search_list param past 200 when using diskann index

### ISSUE THREAD

**Issue by filereadAndrew** [milvus-io/milvus#23672]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.5
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2):v2.2.6
- OS(Ubuntu or CentOS): ubuntu
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When using the following search parameters with the diskann index enabled

search_params = {
    "metric_type": "L2",
    "params": {"k": 200, "search_list":201},
}

I get the following error
`RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:
attempt #1:code: UnexpectedError, error: fail to Search, QueryNode ID=3, reason=Search 6 failed, reason [UnexpectedError] Assert "query_config.search_list_size <= std::max(uint32_t(topk * 10), uint32_t(kSearchListMaxValue1)) && query_config.search_list_size <= uint32_t(kSearchListMaxValue2)" at [/go/src/github.com/milvus-io/milvus/internal/core/src/index/VectorDiskIndex.cpp:122](https://file+.vscode-resource.vscode-cdn.net/go/src/github.com/milvus-io/milvus/internal/core/src/index/VectorDiskIndex.cpp:122)
 => search_list should be less than max(topk*10, 200) and less than 65535 err %!w(<nil>)
attempt #2:context canceled
)>, <Time:{'RPC start': '2023-04-24 16:12:21.376767', 'RPC error': '2023-04-24 16:12:22.069168'}>`

Setting the value for search_list back to 200 seems to work just fine. 

### Expected Behavior

Expected behavior as shown in the docs here https://milvus.io/docs/disk_index.md
should be that we can set search_list value up to min(k*10, 65535) 

### Steps To Reproduce

```markdown
Spin up a cluster of milvus. Enable DISKANN indexing. Upsert vectors. Search for said vectors with above mentioned parameters
```

### Milvus Log

RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:
attempt #1:code: UnexpectedError, error: fail to Search, QueryNode ID=3, reason=Search 6 failed, reason [UnexpectedError] Assert "query_config.search_list_size <= std::max(uint32_t(topk * 10), uint32_t(kSearchListMaxValue1)) && query_config.search_list_size <= uint32_t(kSearchListMaxValue2)" at [/go/src/github.com/milvus-io/milvus/internal/core/src/index/VectorDiskIndex.cpp:122](https://file+.vscode-resource.vscode-cdn.net/go/src/github.com/milvus-io/milvus/internal/core/src/index/VectorDiskIndex.cpp:122)
 => search_list should be less than max(topk*10, 200) and less than 65535 err %!w(<nil>)
attempt #2:context canceled
)>, <Time:{'RPC start': '2023-04-24 16:12:21.376767', 'RPC error': '2023-04-24 16:12:22.069168'}>

**Comment by yanliang567** [comment]
/assign @binbinlv 
please help to check our e2e tests.
/unassign

**Comment by yanliang567** [comment]
this was fixed on milvus 2.2.14
